test/integration/content-explorer/ContentExplorer.int.test.js

```diff
@@ -220,15 +218,49 @@ describe('ContentExplorer', () => {
                 .find(helpers.getSelector(gridViewClass))
                 .should('have.length', 0);
         });
+    });
 
-        it('Should switch to grid view', () => {
+    describe('Grid View', () => {
+        beforeEach(() => {
+            helpers.load({ features: gridViewOn });
             helpers.getViewModeChangeButton().click();
+        });
+
+        it('Should switch to grid view', () => {
             cy.getByTestId('content-explorer')
                 .find(helpers.getSelector(listViewClass))
                 .should('have.length', 0);
             cy.getByTestId('content-explorer')
                 .find(helpers.getSelector(gridViewClass))
                 .should('have.length', 1);
         });
+
+        it('Should open and close share modal', () => {
+            cy.getByAriaLabel(localize('be.share')).should('have.length', 0);
+            cy.contains(localize('be.share')).click();
+            cy.getByAriaLabel(localize('be.share')).should('have.length', 1);
+            helpers.getCloseButton().click();
+            cy.getByAriaLabel(localize('be.share')).should('have.length', 0);
+        });
+
+        it('Should open and close rename modal', () => {
+            cy.getByAriaLabel(localize('be.rename')).should('have.length', 0);
+            helpers
+                .getAllMoreOptionsButtons()
+                .eq(0)
+                .click();
+            helpers.getRenameButton().click();
+            cy.getByAriaLabel(localize('be.rename')).should('have.length', 1);
+            helpers.getCancelButton().click();
+            cy.getByAriaLabel(localize('be.rename')).should('have.length', 0);
+        });
+
+        it('Should open and close preview', () => {
+            cy.getByAriaLabel(localize('be.preview')).should('have.length', 0);
+            cy.contains('Sample 3D.box3d').click();
```

Comment:
Can we instead click the first file in the list to avoid selecting on a specific file name?